### PR TITLE
spawn server (Windows support for external plugins)

### PR DESCRIPTION
**Pull request by ktsaou**
The goals of this PR are:

1. Eliminate all runtime fork() calls from Netdata. This is needed to properly support Windows.
2. Create a new spawn server that works properly on POSIX systems.
3. Using the same spawn server API to implement a native Windows way for spawning external plugins in Windows.

TODO List:

- [x] implement the new POSIX spawn server.
- [x] implement a Windows native version of the spawn server API, to allow external plugins under Windows.
- [x] use the new spawn server for switching name spaces in local-sockets. This enables the efficient use of the NETLINK kernel API (via libmnl), which also enables detecting sockets that work under both IPv4 and IPv6 (using the `IPV6_ONLY` flag), and enables reception of socket delay timings in network viewer.
- [x] use the new sockets detection in `local-listeners`.
- [x] use the new sockets detection in `network-viewer.plugin`.
- [x] use the new spawn server for plugins.d plugins.
- [x] use the new spawn server for claiming.
- [x] use the new spawn server for `tc-qos-helper`.
- [x] use the new spawn server in analytics.
- [x] use the new spawn server in cgroup.plugin.
- [x] use the new spawn server in buildinfo.
- [x] use the new spawn server for agent alert notifications.
- [x] netdata logger now logs `winerror` (the corresponding errno in windows).

CLEANUP:

- [x] Remove the old spawn server
- [x] Remove old netdata_popen()
- [x] Simplify signal handling in Netdata
- [x] Remove the need for `waitid()` (does not exist in cygwin/msys)

### CODE REVIEW

**Comment by ktsaou** [comment]
@ilyam8 I cannot run the tests. Please help verify this works.

For local-listeners, the report will be `TCP46` and `UDP46`.
For network-viewer, the protocol will be `tcp46` and `udp46`.

**Comment by ktsaou** [comment]
It did not work. It seems like we cannot open a socket by opening the fd from /proc.

However, it works via netlink. So, I enabled libmnl by default. It is slower, but now we can have latency information (round trip time) and also total number of retransmits on the sockets.

Since this is now using netlink, I think we need to test it works properly for containers too.

**Comment by ktsaou** [comment]
Do not merge! The multi-threaded fork() is a bad idea. malloc locks are forked locked and the forked processes end up in deadlock.

We need a spawn server for this work correctly!

File: src/libnetdata/safe_fork/safe_fork.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "safe_fork.h"
+
+void string_safe_fork_before(void);
+void string_safe_fork_after(void);
+
+pid_t safe_fork(void) {
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply src/libnetdata/safe_fork/safe_fork.c]: removed

**Comment by stelfrag** [comment]
in freebsd the spawn server is listed as `netdata` eg PID 19063 below

```
netdata 19062   0.0  0.4  120760 35184  -  IN   01:06      0:00.74 /usr/sbin/netdata -P /var/run/netdata/netdata.pid
netdata 19063   0.0  0.2   28132 14912  -  SN   01:06      0:00.01 /usr/sbin/netdata -P /var/run/netdata/netdata.pid
```

**Comment by ktsaou** [comment]
fixed the proc title for freebsd too.

**Comment by stelfrag** [comment]
> fixed the proc title for freebsd too.

Looks good
```
53227 netdata     68  19  110M 34088 S   0.0  0.4  0:00.98 └─ /usr/sbin/netdata -P /var/run/netdata/netdata.pid
53228 netdata     52  19 28132 14428 S   0.0  0.2  0:00.01    └─ netdata: spawn-plugins
53272 netdata     50  19 1270M 60276 S   0.0  0.7  0:00.18       ├─ /usr/libexec/netdata/plugins.d/go.d.plugin 1
53273 root        50  19 33060 10572 S   0.0  0.1  0:00.07       └─ /usr/libexec/netdata/plugins.d/apps.plugin 1
```

**Pull request merged**